sdk/core/azure-core/azure/core/async_paging.py

```diff
@@ -70,13 +71,15 @@ async def __anext__(self) -> ReturnType:
 class AsyncPageIterator(AsyncIterator[AsyncIterator[ReturnType]]):
     def __init__(
         self,
-        get_next: Callable[
+        get_next: Optional[Callable[
             [Optional[str]], Awaitable[ResponseType]
-        ],
-        extract_data: Callable[
+        ]] = None,
+        extract_data: Optional[Callable[
             [ResponseType], Awaitable[Tuple[str, AsyncIterator[ReturnType]]]
-        ],
+        ]] = None,
         continuation_token: Optional[str] = None,
+        paging_method=None,
```

Comment:
I assume that this is an `Optional[str]`? Did we consider overloading the get_next to take a `str | Callable`?

Reply:
Sorry it took me so logn to address. this comment. `paging_method` is an optional `PagingMethod` class, I've added type hints for this. `get_next` is still a `callable`, we've just made it optional, since passing in `get_next` is the older way of paging